configurevalidator/configurevalidator.go

```diff
@@ -0,0 +1,152 @@
+package configurevalidator
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+
+)
+
+var profile string
+var NumberRegion int
+var regions map[string]string
+var numbers[14]string
+var file os.File
+var usr *user.User
+var path string
+var exist bool
+
+func GetExist() bool{
+	return exist
+}
+
+func MakeArrayRegions () [14]string {
+	numbers[0]="us-east-2"
+	numbers[1]="us-east-1"
+	numbers[2]="us-west-1"
+	numbers[3]="us-west-2"
+	numbers[4]="ap-south-1"
+	numbers[5]="ap-northeast-2"
+	numbers[6]="ap-southeast-1"
+	numbers[7]="ap-southeast-2"
+	numbers[8]="ap-northeast-1"
+	numbers[9]="ca-central-1"
+	numbers[10]="eu-central-1"
+	numbers[11]="eu-west-1"
+	numbers[12]="eu-west-2"
+	numbers[13]="sa-east-1"
+
+	return numbers
+
+}
+
+func ShowRegions( ) {
+numbers=MakeArrayRegions()
+	fmt.Println("New configuration file. Choose region:")
+	for i:=0; i<len(numbers);i++{
+		fmt.Println("Number ",i," region ",numbers[i])
+	}
+	for !SetRegions(){
+		SetRegions()
+	}
+
+	fmt.Println("Profile: ")
+	for !SetProfile(){
+		SetProfile()
+	}
+
+	}
+
+func SetRegions( ) (ok bool){
+	//get number from the console
+	fmt.Scan(&NumberRegion)
+	if NumberRegion>=0 && NumberRegion<14 {
+	fmt.Println("Your region is: " + numbers[NumberRegion])
+	ok=true
+	return ok
+} else {
+	ok=false
+	fmt.Println("Invalid region, try again")
+	return ok
+}
+}
+
+func SetProfile() (ok bool){
+	//get name of the profile
+	fmt. Scan(&profile)
+	if profile!="" {
+		fmt.Println("Your profile is: " + profile)
+		ok=true
+		return ok
+	} else {
+		ok=false
+		fmt.Println("Invalid profile, try again")
+		return ok
+	}
+}
+
+func MakePath() string{
+	usr, err := user.Current()
+	if err!=nil {
+		fmt.Println("Error-path")
+		return ""
+	}
+	 path:= usr.HomeDir
+	 return path
+}
+
+func MakeYaml() {
+	//make file with region and profile
+	file, err:=os.Create(MakePath()+"/.config/perun/main.yaml")
+
+	if err !=nil {
+		fmt.Println("Error- file")
+		return
+	}
+
+	defer file.Close()
+	_,err=file.WriteString("Default profiles: "+profile +"\n")
+	_,err=file.WriteString("Default region: "+numbers[NumberRegion]+"\n")
+	_,err=file.WriteString("SpecificationURL: \n")
+
+	for j:=0; j<len(numbers);j++{
+		_,err=file.WriteString("  "+numbers[j]+":"+" "+regions[numbers[j]]+"\n")
+	}
+	if err!=nil {
+		fmt.Println("Error- write to file")
+	}
+}
+
+func IsFileExist(a bool ){
+exist=a
+}
+
+func GetPath()string{
+	configurePath:=MakePath()+"/.config/perun/main.yaml"
+	return configurePath
+}
+
+func MakeMapRegion ()  map[string]string{
```

Comment:
We do not want to hardcode these URL's. They can change over time and keeping them inside configuration file let user to update URL when new version of resource specification come out.

Reply:
Should I do new file only with regions and URL's?

Reply:
Declare it like this outside any function in this package:

```
var resourceSpecificationUrls = map[string]string {
	"us-east-2": "https://dnwj8swjjbsbt.cloudfront.net",
        ...
};
```
It would be enough to perform autoconfiguration for now.

@afronski It would be really cool if Perun will always use latest resource specification. What do you think about parsing table on [this website](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html)? We can do it in a separate task 😃

Reply:
Is that necessary - I thought that _CloudFront_ distribution does not change per region, so in such URL:

```
https://d2senuesg1djtx.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json
```

`latest` piece points always on the most recent version.

Reply:
My bad, you are right :)